### PR TITLE
[AIDAPP-137]: Prevent Ticket Reopening in Aiding App

### PR DESCRIPTION
**Pull request by ankit-canyon**
### Ticket(s) or GitHub Issue

- https://canyongbs.atlassian.net/browse/AIDAPP-137

### Technical Description

> Describe your changes in detail. This should include technical/architectural decisions and reasoning, if applicable.
> Added allow transition in system service for other status except closed. Added Lock Icon for Closed ticket. Closed ticket cannot be edited.

### Screenshots (if appropriate)

screencapture-test-aidingapp-local-service-requests-9c5d4d4d-ac5b-4af3-9760-4ab5be6030b2-2024-07-01-22_48_28
_______________________________________________

### CODE REVIEW

**Comment by ankit-canyon** [comment]
Hi @Orrison, 

On Lock icon tooltip is not working if button is disabled.

Thank you.

**Review by Orrison — CHANGES_REQUESTED** [review]
Tests are currently failing. It could be because the Edit test is making a Service Request where the status is a classification of `closed` so there is no form. Change the failing edit tests to make sure that the factory creates one with an open classification and that may fix it.

If it doesn't let me know and we can pair on seeing what is going on

File: app-modules/service-management/src/Filament/Resources/ServiceRequestResource/Pages/EditServiceRequest.php [hunk 1]
```diff
@@ -166,4 +166,9 @@ protected function mutateFormDataBeforeFill(array $data): array
 
         return $data;
     }
+
+    // public static function shouldRegisterNavigation(array $parameters = []): bool
+    // {
+    //     dd($parameters['record']);
+    // }
```
Comment (Orrison) [inline app-modules/service-management/src/Filament/Resources/ServiceRequestResource/Pages/EditServiceRequest.php:173]: This commented out code should be removed if it is not needed.

File: app-modules/service-management/src/Observers/ServiceRequestObserver.php [hunk 2]
```diff
@@ -86,6 +86,8 @@ public function saved(ServiceRequest $serviceRequest): void
     {
         CreateServiceRequestHistory::dispatch($serviceRequest, $serviceRequest->getChanges(), $serviceRequest->getOriginal());
 
+        $serviceRequest->load('status');
```
Comment (Orrison) [inline app-modules/service-management/src/Observers/ServiceRequestObserver.php:89]: Did this actually solve the issue we discussed where you are getting the wrong status in the email? If not, let's remove it.

**Review by Orrison — CHANGES_REQUESTED** [review]

File: app-modules/service-management/src/Observers/ServiceRequestObserver.php (hunk 2 above)
Comment (ankit-canyon) [reply app-modules/service-management/src/Observers/ServiceRequestObserver.php]: HI,

Yes it's not solved that issue.

Thank you.

**Review by Orrison — APPROVED** [review]

**Pull request merged**